src/plexil-adapter/OwInterface.cpp

```diff
@@ -310,6 +308,97 @@ static void handle_joint_fault (Joint joint, int joint_index,
   handle_overtorque (joint, msg->effort[joint_index]);
 }
 
+void OwInterface::systemFaultMessageCallback
+(const  ow_faults::SystemFaults::ConstPtr& msg)
+{
+  // Publish all joint information for visibility to PLEXIL and handle any
+  // system-level fault messages.
+  uint64_t msg_val = msg->value;
+
+  for (auto const& entry : systemErrors){
+    string key = entry.first;
+    uint64_t value = entry.second.first;
+    bool b = entry.second.second;
+
+    if (!b && ((msg_val & value) == value)){
+      ROS_ERROR("SYSTEM ERROR: %s", key.c_str() );
+      systemErrors[key].second = true;
+    }
+    else if (b && ((msg_val & value) != value)){
+      ROS_INFO("RESOLVED SYSTEM ERROR: %s", key.c_str() );
+      systemErrors[key].second = false;
```

Comment:
It does seem that you have a function `checkFaultMessages` that does the exact behavior which you applied in other parts but not .. is there a reason why you couldn't use that same function here?

Reply:
it's because that function accepts a uint32 where the system faults expects a uint64. If not for that there could be more optimization I think.

Reply:
See my notes on refactoring `checkFaultMessages` to use templates